libbeat/common/kubernetes/util.go

```diff
@@ -36,27 +35,15 @@ const defaultNode = "localhost"
 // GetKubernetesClient returns a kubernetes client. If inCluster is true, it returns an
 // in cluster configuration based on the secrets mounted in the Pod. If kubeConfig is passed,
 // it parses the config file to get the config required to build a client.
-func GetKubernetesClient(inCluster bool, kubeConfig string) (client *k8s.Client, err error) {
-	if inCluster == true {
```

Comment:
it sems you don't use `inCluster` anymore, how would this be used now? We would need to update the comment and some docs

Reply:
the official kube client deduces if it runs in cluster or not and hence to get a kube client we dont really need it. 

```
if kubeconfigPath == "" && masterUrl == "" {
		klog.Warningf("Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.")
		kubeconfig, err := restclient.InClusterConfig()
		if err == nil {
			return kubeconfig, nil
		}
		klog.Warning("error creating inClusterConfig, falling back to default config: ", err)
	}
```

if kubeconfig path is provided then it just tries to create an out of cluster configuration.

That said. let me fix the discover node to just follow the same thing.